sdk/core/Azure.Core.Experimental/tests/DynamicJsonTests.cs

```diff
@@ -278,6 +278,47 @@ public void CanMakeChangesAndAddNewProperty()
             Assert.AreEqual(2, (int)jsonData.Bar);
         }
 
+        [Test]
+        public void CanCheckOptionalProperty()
+        {
+            dynamic json = GetDynamicJson(@"
+                {
+                  ""Foo"" : ""foo""
+                }");
+
+            // Property is present
+            Assert.IsFalse((string)json.Foo == null);
+
+            // Property is absent
+            Assert.IsTrue((string)json.Bar == null);
+        }
+
+        [Test]
+        public void CanCheckOptionalPropertyWithChanges()
+        {
+            dynamic json = GetDynamicJson(@"
+                {
+                  ""Foo"" : ""foo"",
+                  ""Bar"" : {
+                    ""A"" : ""a""
+                  }
+                }");
+
+            // Add property Baz
+            json.Baz = "baz";
+
+            // Remove property A
+            json.Bar = new { B = "b" };
+
+            // Properties are present
+            Assert.IsFalse((string)json.Foo == null);
+            Assert.IsFalse((string)json.Bar.B == null);
+            Assert.IsFalse((string)json.Baz == null);
```

Comment:
What happens if we access a property on a non-existent property? e.g. `json.Baz.Fizz`?